### PR TITLE
Implement shared implementation of StringType.getLocalizedText

### PR DESCRIPTION
**Pull request by ndegwamartin**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Required to close [PR#927](https://github.com/google/android-fhir/pull/927)

**Description**
Refactored to move the StringType.getLocalizedText [here](https://github.com/google/android-fhir/blob/0691f7500e93ed8e0bfb2f3c2e55b86f7c4856bd/datacapture/src/main/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponents.kt#L64) to the common library for reuse across the different SDK modules

**Alternative(s) considered**
See [comment here](https://github.com/google/android-fhir/pull/927#pullrequestreview-824464509)

**Type**
Choose one: (Other)

**Screenshots (if applicable)**
See [PR#927](https://github.com/google/android-fhir/pull/927)

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Comment by Tarun-Bhardwaj** [comment]
@maimoonak , could you please review this PR?

File: common/src/main/java/com/google/android/fhir/MoreTypes.kt [hunk 1]
```diff
@@ -89,3 +91,7 @@ private fun clearTimeFromDateValue(dateValue: Date): Date {
   calendarValue.set(Calendar.MILLISECOND, 0)
   return calendarValue.time
 }
+
+fun StringType.getLocalizedText(lang: String = Locale.getDefault().toLanguageTag()): String? {
+  return getTranslation(lang) ?: getTranslation(lang.split("-").first()) ?: value
```
Comment (maimoonak) [inline common/src/main/java/com/google/android/fhir/MoreTypes.kt:96]: we need to use firstOrNull instead of first

**Review by maimoonak — APPROVED** [review]

File: common/src/test/java/com/google/android/fhir/MoreTypesTest.kt [hunk 2]
```diff
@@ -238,4 +242,76 @@ class MoreTypesTest {
       value.compareTo(otherValue)
     }
   }
+
+  @Test
+  fun localizedText_swahiliTranslation_shouldReturnTranslatedText() {
+
```
Comment (jingtang10) [inline common/src/test/java/com/google/android/fhir/MoreTypesTest.kt:248]: remove empty line

File: common/src/test/java/com/google/android/fhir/MoreTypesTest.kt [hunk 3]
```diff
@@ -238,4 +242,76 @@ class MoreTypesTest {
...
+          }
+        )
+      }
+
+    assertThat(displayElement.getLocalizedText()).isEqualTo("Mwanaume")
+  }
+
+  @Test
+  fun localizedText_swahiliTranslationWithLocale_shouldReturnTranslatedText() {
+
```
Comment (jingtang10) [inline common/src/test/java/com/google/android/fhir/MoreTypesTest.kt:266]: remove empty line

File: common/src/test/java/com/google/android/fhir/MoreTypesTest.kt [hunk 4]
```diff
@@ -238,4 +242,76 @@ class MoreTypesTest {
...
+          }
+        )
+      }
+
+    assertThat(displayElement.getLocalizedText()).isEqualTo("Mwanamke")
+  }
+
+  @Test
+  fun localizedText_noMatchingLocaleTranslation_shouldReturnDefaultText() {
+
```
Comment (jingtang10) [inline common/src/test/java/com/google/android/fhir/MoreTypesTest.kt:284]: remove empty line

File: common/src/test/java/com/google/android/fhir/MoreTypesTest.kt [hunk 5]
```diff
@@ -238,4 +242,76 @@ class MoreTypesTest {
...
+          }
+        )
+      }
+
+    assertThat(displayElement.getLocalizedText()).isEqualTo("Woman")
+  }
+
+  @Test
+  fun localizedText_localeTagWithCountry_shouldReturnTranslatedText() {
+
```
Comment (jingtang10) [inline common/src/test/java/com/google/android/fhir/MoreTypesTest.kt:302]: remove empty line

**Review by jingtang10 — APPROVED** [review]
just a few nitpicks - once addressed please feel free to merge!

thanks so much!

**Pull request merged**